### PR TITLE
Fix clearing of Version fields via REST API

### PR DESCRIPTION
**Pull request by dregad**
Fixes [#25407](https://mantisbt.org/bugs/view.php?id=25407)

Also fixes the error message when an invalid version is provided ([#34410](https://mantisbt.org/bugs/view.php?id=34410)).

### CODE REVIEW

File: api/soap/mc_issue_api.php [hunk 1]
```diff
@@ -1010,9 +1010,12 @@ function mc_issue_update( $p_username, $p_password, $p_issue_id, stdClass $p_iss
 	$t_category = isset( $p_issue['category'] ) ? $p_issue['category'] : null;
 	$t_category_id = mci_get_category_id( $t_category, $t_project_id );
 
-	$t_version_id = isset( $p_issue['version'] ) ? mci_get_version_id( $p_issue['version'], $t_project_id, 'version' ) : 0;
-	$t_fixed_in_version_id = isset( $p_issue['fixed_in_version'] ) ? mci_get_version_id( $p_issue['fixed_in_version'], $t_project_id, 'fixed_in_version' ) : 0;
-	$t_target_version_id = isset( $p_issue['target_version'] ) ? mci_get_version_id( $p_issue['target_version'], $t_project_id, 'target_version' ) : 0;
+	$fn_get_version_id = function( $p_version, int $p_project_id, string $p_field ): int {
+		return isset( $p_version ) ? mci_get_version_id( $p_version, $p_project_id, $p_field ) : 0;
```
Comment (vboctor) [inline api/soap/mc_issue_api.php:1014]: To empty a version field set its id to 0. I added a note to the bug with more details.

**Review by vboctor — CHANGES_REQUESTED** [review]

File: api/soap/mc_issue_api.php (hunk 1 above)
Comment (dregad) [reply api/soap/mc_issue_api.php:1014]: I replied in the tracker https://mantisbt.org/bugs/view.php?id=25407#c68848

**Comment by dregad** [comment]
> Also fixes the error message when an invalid version is provided ([#34410](https://mantisbt.org/bugs/view.php?id=34410)).

I have merged this minor fix in 2.26 branch (see 4859fc1fbcf7547e057cc4520a880a2b56231b55), without waiting for this PR's review

**Comment by dregad** [comment]
Without feedback following my note https://mantisbt.org/bugs/view.php?id=25407#c68929, I'm going to merge this sometime this week.

**Pull request merged**